### PR TITLE
feat(order-entry): keep microbiology details on qualifying orders only

### PR DESCRIPTION
**Pull request by pmanko**
## Stack

- Base: [R13 first-isolate policy #4124](https://github.com/DIGI-UW/OpenELIS-Global-2/pull/4124)
- Next native stacked OGC-782 roadmap slice; intentionally targets the R13 branch.

## Behavior delivered

- One rule decides microbiology eligibility everywhere: a configured culture-workflow test, or an explicitly selected Microbiology program as the documented fallback. The details section, Program derivation, readiness and the submitted payload all defer to it.
- The server decides from the catalog and ignores what the client claims about a submitted test.
- A routine order never stores microbiology details, even when a client sends them.
- An order that stops qualifying before collection discards the details it captured; details already recorded against a case are untouched.
- Discarding details clears them outright, so a re-added culture test starts at the clinical purpose the backend requires.
- Number of sets appears on every microbiology order, bounded 1–10, with helper text separating it from specimen quantity.

## Intentional scope

- Recoverable inline validation, the completed/next-action state, and canonical navigation are R15/R16 (#4193 and following).
- The order-entry rulings and the remaining remediation steps are recorded in `specs/782-ogc-782-microbiology-mvp-spec/tasks.md`.

## Validation

- Backend: 6 eligibility integration tests, including catalog-not-payload, the program fallback exercised with the real empty-XML requested-stage payload, and draft discard. Each new test was inverted against the code it guards.
- Frontend: order, microbiology and form-model suites pass on this branch alone (43 files, 314 tests).
- Full backend suite 6206 tests green; full frontend suite 226 files green.
- Six order-entry Playwright journeys pass against a WAR verified to contain this code.
- `DIGI-UW/code-qa`: simplicity verdict lean; coverage inversions traced; commit hygiene clean.

## Review target

- AMR clinical order entry (Enter Order).
- Roadmap row R14 remains `[*]` until required checks are green at this head.

### CODE REVIEW

File: src/main/java/org/openelisglobal/sample/service/SamplePatientEntryServiceImpl.java [hunk 1]
```diff
@@ -634,6 +636,42 @@ private int normalizeLabelQuantity(Integer quantity) {
...
+     */
+    private boolean isMicrobiologyOrder(SamplePatientUpdateData updateData) {
+        if (microOrderRoutingService == null) {
+            return false;
+        }
+        // The submitted collection carries id-only tests, so the catalog attributes
+        // that decide the workflow are read from the persisted test.
+        List<Test> orderedTests = new ArrayList<>();
+        if (updateData.getSampleItemsTests() != null) {
+            for (SampleTestCollection sampleTestCollection : updateData.getSampleItemsTests()) {
```
[the remark this answers is not in the document]
Comment (pmanko) [reply src/main/java/org/openelisglobal/sample/service/SamplePatientEntryServiceImpl.java:657]: Fixed in this stack: `saveOrderEntry` sends an empty `sampleXML`, so eligibility now also reads the tests named in `requestedSampleTypes` rather than treating an empty sample-items collection as a routine order. `requestedStageQualifiesFromRequestedSampleTypesWhenNoProgramIsSelected` reproduced the loss first.

File: frontend/src/components/order/OrderContext.jsx [hunk 2]
```diff
@@ -623,9 +626,9 @@ export const OrderProvider = ({ children, workflowType = "clinical" }) => {
         sampleOrderItems: buildSubmissionSampleOrderItems(
           orderData.sampleOrderItems,
         ),
-        microbiologyOrderDetail: buildSubmissionMicrobiologyOrderDetail(
-          orderData.microbiologyOrderDetail,
-        ),
+        ...(submittedMicrobiologyOrderDetail
+          ? { microbiologyOrderDetail: submittedMicrobiologyOrderDetail }
+          : {}),
```
[the remark this answers is not in the document]
Comment (pmanko) [reply frontend/src/components/order/OrderContext.jsx:631]: Fixed: both submit paths assign `microbiologyOrderDetail` explicitly rather than spreading a conditional object, and `JSON.stringify` drops the key when the builder returns `undefined`.

File: src/main/java/org/openelisglobal/microbiology/daoimpl/MicroCaseOrderDetailDAOImpl.java [hunk 3]
```diff
@@ -37,6 +37,14 @@ public MicroCaseOrderDetail getDraftBySampleId(String sampleId) {
         return query.uniqueResultOptional().orElse(null);
     }
 
+    @Override
+    public void deleteDraftBySampleId(String sampleId) {
+        MicroCaseOrderDetail draft = getDraftBySampleId(sampleId);
+        if (draft != null) {
+            delete(draft);
```
[the remark this answers is not in the document]
Comment (pmanko) [reply src/main/java/org/openelisglobal/microbiology/daoimpl/MicroCaseOrderDetailDAOImpl.java:44]: Fixed in this stack: losing eligibility now retires the draft with an actor and a timestamp (`discarded_at` / `discarded_by`, Liquibase `091`) instead of hard-deleting it, and re-qualifying revives that row. `micro_case_activity` requires a `case_id`, which a pre-case draft cannot supply, so the record lives on the draft itself. Covered by a service test and `MicrobiologyOrderDraftDiscardAuditLiquibaseRollbackTest`.

File: src/main/java/org/openelisglobal/sample/service/SamplePatientEntryServiceImpl.java [hunk 4]
```diff
@@ -634,6 +636,42 @@ private int normalizeLabelQuantity(Integer quantity) {
...
+                    }
+                    Test catalogTest = testService.get(submittedTest.getId());
+                    if (catalogTest != null) {
+                        orderedTests.add(catalogTest);
+                    }
+                }
+            }
+        }
+        return microOrderRoutingService.isMicrobiologyOrder(orderedTests, microbiologyProgramSelected(updateData));
+    }
```
[the remark this answers is not in the document]
Comment (pmanko) [reply src/main/java/org/openelisglobal/sample/service/SamplePatientEntryServiceImpl.java:673]: Fixed: catalog tests are loaded once per id into a map that eligibility and the requested-types fallback share.

File: frontend/src/components/order/OrderContext.jsx [hunk 5]
```diff
@@ -861,9 +864,14 @@ export const OrderProvider = ({ children, workflowType = "clinical" }) => {
           // Include per-sample vector observations merged above.
           environmentalFields: envFields,
         }),
-        microbiologyOrderDetail: buildSubmissionMicrobiologyOrderDetail(
-          orderData.microbiologyOrderDetail,
-        ),
+        ...(buildSubmittedMicrobiologyOrderDetail(orderData, samples)
```
[the remark this answers is not in the document]
Comment (pmanko) [reply frontend/src/components/order/OrderContext.jsx:867]: Fixed: the helper is computed once per submit path.
